### PR TITLE
Repo list + checks for prettier config for each github repository

### PR DESCRIPTION
**Pull request by ippsav**

### CODE REVIEW

File: packages/backend/src/types/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+export interface IGithubAPIRepo {
+  owner: string;
+  name: string;
+  full_name: string;
+  language: string;
+  contents_url: string;
+}
+
+export interface IRepoResponse extends Map {
```
Comment (ippsav) [inline packages/backend/src/types/index.ts:24]: The purpose of extending this interface to a map is to be able to get values the way we can get it from any map( map[string])

File: packages/backend/src/handlers/repository.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            Authorization: `token ${accessToken}`,
+            Accept: 'application/vnd.github.v3+json',
+          },
+        });
+        const directoryContent = await directoryContentResponse.json<IGithubFileContent[]>();
+        if (packageConfig.prettier) {
+          repository.prettier = true;
+        }
+        for (const [key, value] of Object.entries(configFiles)) {
+          if (!repository[key]) {
```
Comment (ippsav) [inline packages/backend/src/handlers/repository.ts:87]: Like this

File: packages/backend/src/handlers/repository.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        const directoryContentResponse = await fetch(`${contentUri}/`, {
+          headers: {
+            Authorization: `token ${accessToken}`,
+            Accept: 'application/vnd.github.v3+json',
+          },
+        });
+        const directoryContent = await directoryContentResponse.json<IGithubFileContent[]>();
+        if (packageConfig.prettier) {
+          repository.prettier = true;
+        }
```
Comment (ippsav) [inline packages/backend/src/handlers/repository.ts:85]: first checking if the package.json contains prettier config

File: packages/backend/src/handlers/repository.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+              if ((value as string[]).includes(entry.name)) {
+                repository[key] = true;
+                break;
+              }
+            }
+          }
+        }
+        responseBody = [...responseBody, repository];
+      }
+    }
```
Comment (ippsav) [inline packages/backend/src/handlers/repository.ts:98]: if not then looking at the root folder if there is any files with the name of a prettier config file

File: packages/backend/src/handlers/repository.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    '.prettierrc.json',
+    '.prettierrc.yml',
+    '.prettierrc.json5',
+    '.prettierrc.js',
+    '.prettierrc.cjs',
+    'prettier.config.js',
+    'prettier.config.cjs',
+    '.prettierrc.toml',
+  ],
+};
```
Comment (ippsav) [inline packages/backend/src/handlers/repository.ts:23]: which are these files

File: packages/frontend/src/App.tsx [hunk 6]
```diff
@@ -58,7 +59,7 @@ const App = () => {
           isAuthenticated={isAuthenticated}
           path="/repo-list"
           redirectTo="/"
-          component={Table}
+          component={Repostiories}
```
Comment (frolovdev) [inline packages/frontend/src/App.tsx:62]: RepositoryList

File: packages/frontend/src/pages/Login.tsx [hunk 7]
```diff
@@ -57,7 +57,7 @@ export const Login: FC = () => {
         Get your dashboard
       </div>
       <div className="mt-4">
-        <Table />
+        <Table isPlaceholder={true} />
```
Comment (frolovdev) [inline packages/frontend/src/pages/Login.tsx:60]: same

```
<Table isPlaceholder />
```

File: packages/backend/src/index.ts [hunk 8]
```diff
@@ -1,83 +1,13 @@
...
-  return new Response('status: OK', {
-    status: 200,
-    headers: {
-      'Set-Cookie': `qid=${access_token}; SameSite=Lax; HttpOnly; Path=/${
-        ctx.env.ENVIRONMENT === 'production' ? `; Secure; Domain=${ctx.env.DOMAIN};` : ''
-      }`,
-    },
-  });
-};
+import { authenticate, isAuthenticated } from './handlers/auth';
```
Comment (frolovdev) [inline packages/backend/src/index.ts:3]: handlers/auth.handler.ts

File: packages/backend/src/handlers/repository.ts [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+      ) {
+        const repository: IRepoResponse = {
+          owner: repo.owner,
+          name: repo.name,
+          fullName: repo.full_name,
+          contentUrl: repo.contents_url,
+          language: repo.language,
+          prettier: false,
+          eslint: false,
+          jest: false,
```
Comment (frolovdev) [inline packages/backend/src/handlers/repository.ts:56]: could we refactor this code a little bit, so the handler should only handle I/O and let's introduce a service layer, where logic related to prettier will be encapsulated

File: packages/backend/src/handlers/repository.ts [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        };
+        const contentUri = repo.contents_url.slice(0, repo.contents_url.lastIndexOf('/'));
+        const contentResponse = await fetch(`${contentUri}/package.json`, {
```
Comment (frolovdev) [inline packages/backend/src/handlers/repository.ts:59]: at start you make a one http request, and after thay in mostly anyways you will do another HTTP request (imagine if we will need to have more integrations to analyze), so it looks much easier to make one request without any repetitions to root (/) and retrieve content from it

Comment (ippsav) [reply packages/backend/src/handlers/repository.ts:59]: it s impossible to make only one request: 
1- we need to fetch the package.json in a single request to get it s content if we fetch the whole directory we get only basic pieces of information about the files inside, like name, and we need it s content to check if the config for the specific dep like prettier is written directly to package.json
2- we fetch the root folder structure to see if we have the config file for the specific dep if we do not have it inside package.json

File: packages/backend/src/types/index.ts (hunk 1 above)
Comment (frolovdev) [reply packages/backend/src/types/index.ts:24]: I guess it's useless, don't extend and monkey patch the classic map interface just use .get method or use separate function

File: packages/backend/src/handlers/repository.ts (hunk 2 above)
Comment (frolovdev) [reply packages/backend/src/handlers/repository.ts:87]: looks bad to me

File: packages/backend/src/handlers/repository.handler.ts [hunk 11]
```diff
@@ -0,0 +1,81 @@
+import { githubApi } from 'src/config';
+import { extractCookie } from 'src/cookie';
+import { isPrettier } from 'src/services/prettier.service';
```
Comment (frolovdev) [inline packages/backend/src/handlers/repository.handler.ts:3]: let's avoid absolute imports 🙏

File: packages/backend/src/handlers/repository.handler.ts [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+          name: repo.name,
+          fullName: repo.full_name,
+          contentUrl: repo.contents_url,
+          language: repo.language,
+          prettier: isPrettier({ directoryContent, packageConfig }),
+          eslint: false,
+          jest: false,
+        };
+        responseBody = [...responseBody, repository];
+      }
```
Comment (frolovdev) [inline packages/backend/src/handlers/repository.handler.ts:66]: split the coe to smaller functions 🙏

**Review by frolovdev — APPROVED** [review]

**Pull request merged**